### ISSUE TITLE
Trackpad buttons doesn't work on Hp bs171nia synaptics

### ISSUE THREAD

**Issue by PriNcEFuL** [acidanthera/bugtracker#1108]
so after using appleps2smarttouchpad for long time without proper gestures i decided to use voodoops2 now all gesture and trackpad works only physical buttons doesn't work

**Comment by wsy** [comment]
I got similar problem. I'm using OpenCore 0.6.0 and VoodooPS2Trackpad.kext on Lenovo K29 laptop with latest macOS Catalina.

At first I followed the guide and used ProperTree Cmd+Shift+R short cut and got my kexts configured.
In my config.plist, "/EFI/OC/Kexts/VoodooPS2Controller.kext/Contents/Plugins/VoodooPS2Mouse.kext" comes before "/EFI/OC/Kexts/VoodooPS2Controller.kext/Contents/Plugins/VoodooPS2Trackpad.kext".
Then my trackpad got recognized as a mouse and the physical buttons does not work. Simple tap gesture works as left button click.

I tried to adjust their order and put VoodooPS2Trackpad.kext before VoodooPS2Mouse.kext, then I restarted my laptop and the physical buttons still does not work. I opened System Preferences and found my trackpad got recognized as trackpad. Then, for a few seconds, those two physical buttons worked! I thought maybe it needs some kinds of driver reset/re-initialization to work. A few moments later those two buttons stopped working. I went back to System Preference, they worked again for another few seconds. After that, they never worked again even if I went back to System Preference.

Later I'm gonna change my OpenCore to a debug version and see if the trackpad button will work after restart.

**Comment by HQuest** [comment]
Assuming you both have PS2 compatible trackpads/touchpads, are you making use of VoodooInput.kext as well? It is a requirement for some features of the VoodooPS2/VoodooI2C.

**Comment by PriNcEFuL** [comment]
> Assuming you both have PS2 compatible trackpads/touchpads, are you making use of VoodooInput.kext as well? It is a requirement for some features of the VoodooPS2/VoodooI2C.

i use the latest ps2 controller kext which contains 4 kexts and voodooinput is one of them so i use it yeah

**Comment by wsy** [comment]
> Assuming you both have PS2 compatible trackpads/touchpads, are you making use of VoodooInput.kext as well? It is a requirement for some features of the VoodooPS2/VoodooI2C.

Yes, I'm using VoodooInput.kext bundled inside VoodooPS2Controller.kext

**Comment by PriNcEFuL** [comment]
> I got similar problem. I'm using OpenCore 0.6.0 and VoodooPS2Trackpad.kext on Lenovo K29 laptop with latest macOS Catalina.
> 
> At first I followed the guide and used ProperTree Cmd+Shift+R short cut and got my kexts configured.
> In my config.plist, "/EFI/OC/Kexts/VoodooPS2Controller.kext/Contents/Plugins/VoodooPS2Mouse.kext" comes before "/EFI/OC/Kexts/VoodooPS2Controller.kext/Contents/Plugins/VoodooPS2Trackpad.kext".
> Then my trackpad got recognized as a mouse and the physical buttons does not work. Simple tap gesture works as left button click.
> 
> I tried to adjust their order and put VoodooPS2Trackpad.kext before VoodooPS2Mouse.kext, then I restarted my laptop and the physical buttons still does not work. I opened System Preferences and found my trackpad got recognized as trackpad. Then, for a few seconds, those two physical buttons worked! I thought maybe it needs some kinds of driver reset/re-initialization to work. A few moments later those two buttons stopped working. I went back to System Preference, they worked again for another few seconds. After that, they never worked again even if I went back to System Preference.
> 
> Later I'm gonna change my OpenCore to a debug version and see if the trackpad button will work after restart.

i tried reordering the kexts like you did but ended up with not working keyboard and trackpad

**Comment by wsy** [comment]
> i tried reordering the kexts like you did but ended up with not working keyboard and trackpad

Sorry to hear that. I only swapped VoodooPS2Trackpad and VoodooPS2Mouse. I changed nothing on other kexts. I did not change where VoodooInput, VoodooPS2Keyboard and VoodooPS2Controller locates in config.plist
